### PR TITLE
Bump CAPI infra providers versions

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/data/capi-operator/full-providers.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ metadata:
 spec:
   type: infrastructure
   name: azure
-  version: v1.11.5
+  version: v1.16.0
```
Comment (Danil-Grigorev) [inline test/e2e/data/capi-operator/full-providers.yaml:28]:
```suggestion
```
This should be enough since the CAPIProvider version is latest now?

**Review by Danil-Grigorev — DISMISSED** [review]

File: test/e2e/data/capi-operator/full-providers.yaml (hunk 1 above)
Comment (alexander-demicev) [reply test/e2e/data/capi-operator/full-providers.yaml:28]: I guess we can totally remove the version from all CAPIProvider we have in clusterctl config

Comment (alexander-demicev) [reply test/e2e/data/capi-operator/full-providers.yaml:28]: actually let's do it here

**Review by Danil-Grigorev — APPROVED** [review]
Lgtm

**Review by furkatgofurov7 — COMMENTED** [review]
Can we bump Kubeadm Bootstrap/CP versions in the same PR?

**Comment by alexander-demicev** [comment]
@furkatgofurov7 Let's bump them together with CAPI and CAPD later when we switch to a newer version of CAPI

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**